nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -316,15 +316,18 @@ def _wrap_model_for_O2(self):
         args.pop('module')
 
     def get_model_module_list(self):
```

Comment:
@dimapihtar Need your feedback for this change.

This function extracts the `.module` from the `model`; when we wrap a model with `MCoreDDP` or `Float16Module`, the `model` can be accessed via `.module` attribute; however those classes can be combined, so it's possible to have `<obj>.module.module`, thus the recursive call.